File: typescript/session3/td/UnionIter.ts
import { Mot } from "./Mot"

export class UnionIter extends Mot {
    private motGauche : Mot;
    private motDroit : Mot;
    private var_taille : number;

    public constructor(motGauche : Mot, motDroit : Mot) {
        super();
        this.motGauche = motGauche;
        this.motDroit = motDroit;
        this.var_taille = motGauche.taille() + motDroit.taille();
    }

    public casUnion() : boolean {
        return true;
    }

    public gauche() : Mot {
        return this.motGauche;
    }

    public droit() : Mot {
        return this.motDroit;
    }

    public taille() : number {
        return this.var_taille;
    }

    public lettre(): string {
        var i = this.iterateur();
        return i.next().value;
    }

    public reste(): Mot {
        var i = this.iterateur();
        return i.reste();
    }
}

File: java/cours/src/session3/td/src/UnionIter.java
import java.util.Iterator;

public class UnionIter implements Mot{
    private Mot gauche;
    private Mot droit;
    private int taille;

    UnionIter(Mot gauche, Mot droit) {
        this.gauche = gauche;
        this.droit = droit;
        this.taille = gauche.taille() + droit.taille();
    }

    @Override
    public Mot gauche() {
        return gauche;
    }

    @Override
    public Mot droit() {
        return droit;
    }

    @Override
    public boolean casUnion() {
        return true;
    }

    public String toString() {
        return gauche.toString() + droit.toString();
    }

    @Override
    public int taille() {
        return taille;
    }

    @Override
    public char lettre() {
        Iterateur i = this.iterateur();
        return i.next();
    }

    @Override
    public Mot reste() {
        Iterateur i = this.iterateur();
        return i.reste();
    }
}
